src/mapWithFeedback.ts

```diff
@@ -0,0 +1,135 @@
+ * @param initialValue the initial value to start the computation with
+ * @returns An array of successively computed values from the left side of the array.
+ * @signature
+ *    R.mapWithFeedback(fn, initialValue)(array)
+ * @example
+ *    R.pipe([1, 2, 3, 4, 5], R.mapWithFeedback((prev, x) => prev + x, 100)) // => [101, 103, 106, 110, 115]
+ *    R.pipe([1, 2, 3, 4, 5], R.mapWithFeedback.indexed((prev, x, i, array) => prev + x, 100)) // => [101, 103, 106, 110, 115]
+ * @dataLast
+ * @indexed
+ * @pipeable
+ * @category Array
+ */
+export function mapWithFeedback<T extends IterableContainer, Value>(
+  reducer: (previousValue: Value, currentValue: T[number]) => Value,
+  initialValue: Value,
+): (items: T) => StrictOut<T, Value>;
+
+export function mapWithFeedback(): unknown {
+  return purry(
+    mapWithFeedbackImplementation(false),
+    arguments,
+    mapWithFeedback.lazy,
+  );
+}
+
+const mapWithFeedbackImplementation =
+  (indexed: boolean) =>
+  <T, Value>(
+    items: ReadonlyArray<T>,
+    reducer: (
+      previousValue: Value,
+      currentValue: T,
+      index?: number,
+      items?: ReadonlyArray<T>,
+    ) => Value,
+    initialValue: Value,
+  ) => {
+    const implementation = indexed
+      ? mapWithFeedback.lazyIndexed
+      : mapWithFeedback.lazy;
+
+    return _reduceLazy(items, implementation(reducer, initialValue), indexed);
+  };
+
+const lazyImplementation =
+  (indexed: boolean) =>
+  <T, Value>(
+    reducer: (
+      previousValue: Value,
+      currentValue: T,
+      index?: number,
+      items?: ReadonlyArray<T>,
+    ) => Value,
+    initialValue: Value,
+  ): LazyEvaluator<T, Value> => {
+    let previousValue = initialValue;
+    return (value, index, items) => {
+      previousValue = indexed
+        ? reducer(previousValue, value, index, items)
+        : reducer(previousValue, value);
+      return {
+        done: false,
+        hasNext: true,
+        next: previousValue,
+      };
+    };
+  };
+
+export namespace mapWithFeedback {
+  export function indexed<T extends IterableContainer, Value>(
+    items: T,
+    reducer: (
+      previousValue: Value,
+      currentValue: T[number],
+      index: number,
+      items: T,
+    ) => Value,
+    initialValue: Value,
+  ): StrictOut<T, Value>;
+  export function indexed<T extends IterableContainer, Value>(
+    reducer: (
+      previousValue: Value,
+      currentValue: T[number],
+      index: number,
+      items: T,
+    ) => Value,
+    initialValue: Value,
+  ): (items: T) => StrictOut<T, Value>;
+  export function indexed(): unknown {
+    return purry(
+      mapWithFeedbackImplementation(true),
+      arguments,
+      mapWithFeedback.lazyIndexed,
+    );
+  }
+
+  export const lazy = lazyImplementation(false);
+  export const lazyIndexed = _toLazyIndexed(lazyImplementation(true));
+}
+
+type StrictOut<T extends IterableContainer, K> = {
+  -readonly [P in keyof T]: K;
+};
```

Comment:
Pull this up to the top of the file and rename it to Mapped or something, you can even pull this to the _types.ts file and re-use it in both map and here...

The name StrictOut was deliberate to make it stand out against the non-strict version. In this function we don't have a non-strict version so the name doesn't make sense. Also in the past I called all returned types "Out" because i copied these utilities from my own project, now I prefer calling the type by what it does (or more accurately, what it **is**, e.g. First, Grouped, Zipped, etc...)